src/matrices.jl

```diff
@@ -0,0 +1,76 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENCE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    JColumn{T}(d)
+
+Column of ones.
+"""
+struct JColumn{T} <: AbstractVector{T}
+  d::Int
+end
+JColumn(d::Int) = JColumn{Bool}(d)
+Base.size(A::JColumn) = (A.d,)
+Base.IndexStyle(::Type{<:JColumn}) = IndexLinear()
+Base.getindex(A::JColumn{T}, i::Integer) where {T} = one(T)
+
+
+"""
+    JMatrix{T}(d)
+
+Square matrix of ones.
+"""
+struct JMatrix{T} <: AbstractMatrix{T}
+  d::Int
+end
+JMatrix(d::Int) = JMatrix{Bool}(d)
+Base.size(A::JMatrix) = (A.d, A.d)
+Base.IndexStyle(::Type{<:JMatrix}) = IndexCartesian()
+Base.getindex(A::JMatrix{T}, i::Integer, j::Integer) where {T} = one(T)
+
+
+"""
+    FMatrix{T}(d)
+
+`F` matrix, as defined by Aitchison 1986.
+"""
+struct FMatrix{T} <: AbstractMatrix{T}
+  d::Int
+end
+FMatrix(d::Int) = FMatrix{Int}(d)
```

Comment:
Notice that you are not consistent regarding the parameter type. For the Jmatrix you used Bool and here you used Int.

Reply:
The elements in `FMatrix` can assume negative values.

The elements in `GMatrix` are fractional, so `Float64` was chosen as the default. According to the [Performance Tips](https://docs.julialang.org/en/v1/manual/performance-tips/#man-performance-abstract-container) section of the manual, the use of `Real` types should be avoided in containers, but for in my tests `Real` defaulted to `Float64` and performed w/ the same speed.

Reply:
Got it, thank you for clarifying it 💯